validator/src/admin_rpc_service.rs

```diff
@@ -604,6 +611,19 @@ impl AdminRpc for AdminRpcImpl {
         debug!("set_public_tpu_address rpc request received: {public_tpu_addr}");
 
         meta.with_post_init(|post_init| {
+            post_init
+                .cluster_info
+                .my_contact_info()
+                .tpu()
+                .map_err(|err| {
+                    error!(
+                        "The public TPU address isn't being published. \
+                        The node is likely in repair mode. \
+                        See help for --restricted-repair-only-mode for more information. \
+                        {err}"
+                    );
+                    jsonrpc_core::error::Error::internal_error()
+                })?;
```

Comment:
Of course, in this case, these lines of code should be in a separate PR. However, since this is new functionality that only exists at the tip of the master branch, then...
I can move it to a separate PR if it's not acceptable here.